corehq/apps/dump_reload/management/commands/dump_case_data.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from datetime import datetime
+
+from django.db import connections
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.dump_reload.const import DATETIME_FORMAT
+from corehq.form_processor.models import CommCareCaseSQL
+
+
+class Command(BaseCommand):
+    help = "Dump all case data, per case type per shard in csv files"
+
+    def add_arguments(self, parser):
+        parser.add_argument('--domain')
+        parser.add_argument('--case_type')
+        parser.add_argument('--db_conn')
+
+    def ensure_params(self, for_domain, for_case_type, for_db_conn):
+        if for_db_conn not in settings.DATABASES:
+            raise CommandError("Invalid db conn. Please choose from %s" % settings.DATABASES)
+        if for_case_type:
+            available_case_types = (CommCareCaseSQL.objects.using(for_db_conn).
+                                    values_list('type', flat=True).distinct())
+            if for_case_type not in available_case_types:
+                raise CommandError("Unexpected case type {for_case_type} passed. "
+                                   "Should be from the following: {case_types}"
+                                   .format(for_case_type=for_case_type, case_types=available_case_types))
+
+        if for_domain:
+            available_domains = (CommCareCaseSQL.objects.using(for_db_conn).
+                                 values_list('domain', flat=True).distinct())
+            if for_domain not in available_domains:
+                raise CommandError("Domain name {for_domain} not found for any case in db {for_db_conn}".format(
+                    for_domain=for_domain, for_db_conn=for_db_conn
+                ))
```

Comment:
These `distinct` queries might take a long time to run and aren't strictly necessary since if you specify a domain / case type that doesn't exist you just won't get any data.

Reply:
Thats true.. but then just flagging this for the user so that an empty result file is not confused with an error in the script.